### PR TITLE
feat(addon-commerce): add `Decimal` pipe

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
Closes #

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/addon-commerce/pipes/decimal/decimal.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import {inject, INJECTOR, Injector, Pipe} from '@angular/core';
+import {TuiAmountPipe} from '@taiga-ui/addon-commerce/pipes/amount';
+import type {TuiCurrencyVariants} from '@taiga-ui/addon-commerce/types';
+import {TUI_NUMBER_FORMAT} from '@taiga-ui/core/tokens';
+import type {Observable} from 'rxjs';
+import {map, switchMap} from 'rxjs';
+
+import {TUI_DECIMAL_OPTIONS} from './decimal.options';
+
+@Pipe({standalone: true, name: 'tuiDecimal'})
```
Comment (splincode) [inline projects/addon-commerce/pipes/decimal/decimal.pipe.ts:11]: `tuiDecimalPart`?

Comment (splincode) [reply projects/addon-commerce/pipes/decimal/decimal.pipe.ts:11]: image

File: projects/addon-commerce/pipes/decimal/decimal.options.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import type {Provider} from '@angular/core';
+import type {TuiCurrencyVariants} from '@taiga-ui/addon-commerce/types';
+import {tuiCreateToken, tuiProvideOptions} from '@taiga-ui/cdk/utils/miscellaneous';
+
+export interface TuiDecimalOptions {
+    readonly currency: TuiCurrencyVariants;
```
Comment (waterplea) [inline projects/addon-commerce/pipes/decimal/decimal.options.ts:6]: I don't think we need options for that. I believe most of the world places currency before the number and we do not need this pipe in our internal designs.

File: projects/addon-commerce/pipes/decimal/decimal.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+export class TuiDecimalPipe implements PipeTransform {
+    private readonly injector = inject(INJECTOR);
+    private readonly localInjector = Injector.create({
```
Comment (waterplea) [inline projects/addon-commerce/pipes/decimal/decimal.pipe.ts:14]: No need to store it

File: projects/addon-commerce/pipes/amount/amount.pipe.ts [hunk 4]
```diff
@@ -35,7 +35,8 @@ export class TuiAmountPipe implements PipeTransform {
                         : format.precision,
                 });
                 const space =
-                    currencySymbol?.length > 1 || currencyAlign === 'right'
+                    currencySymbol &&
+                    (currencySymbol?.length > 1 || currencyAlign === 'right')
```
Comment (waterplea) [inline projects/addon-commerce/pipes/amount/amount.pipe.ts:39]: Why this change?

Comment (vladimirpotekhin) [reply projects/addon-commerce/pipes/amount/amount.pipe.ts:39]: it was bug, when we added space without currencySymbol but with currencyAlign = 'right'

File: projects/addon-commerce/pipes/decimal/decimal.pipe.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import {inject, INJECTOR, Injector, Pipe} from '@angular/core';
+import {TuiAmountPipe} from '@taiga-ui/addon-commerce/pipes/amount';
+import type {TuiCurrencyVariants} from '@taiga-ui/addon-commerce/types';
+import {TUI_NUMBER_FORMAT} from '@taiga-ui/core/tokens';
+import type {Observable} from 'rxjs';
+import {map, switchMap} from 'rxjs';
+
+@Pipe({standalone: true, name: 'tuiDecimal'})
+export class TuiDecimalPipe implements PipeTransform {
+    private readonly injector = inject(INJECTOR);
```
Comment (waterplea) [inline projects/addon-commerce/pipes/decimal/decimal.pipe.ts:11]: Also unnecessary to store :)

**Review by waterplea — APPROVED** [review]
